src/GafferScene/RenderController.cpp

```diff
@@ -1318,6 +1572,24 @@ void RenderController::updateInternal( const ProgressCallback &callback, const I
 				m_changedGlobalComponents |= CameraOptionsGlobalComponent;
 			}
 			m_globals = globals;
+
+			MotionBlurOptions newOptions( m_globals.get(), m_scene.get() );
+			if( !( newOptions == m_motionBlurOptions ) )
+			{
```

Comment:
FIxed in 6d00f94.

I went perhaps even further then you were suggesting, and move all logic out of MotionBlurOptions.  I'm not sure why I thought it would make things cleaner, but I didn't like how it turned out.